src/freenas/usr/local/libexec/nas/generate_smb4_conf.py

```diff
@@ -936,11 +936,17 @@ def generate_smb4_tdb(client, smb4_tdb):
         return
 
 
-def generate_smb4_conf(client, smb4_conf, role):
+def generate_smb4_conf(client, smb4_conf, role, shares):
     global is_truenas_ha
 
     cifs = Struct(client.call('smb.config'))
 
+    guest_enabled = False
+
+    for share in shares:
+        if share['cifs_guestok']:
+            guest_enabled = True
```

Comment:
we should add a break here to stop searching

or just `if any(filter(lambda x: x['cifs_guestok'], shares)`